### ISSUE TITLE
Inverting Flatlist does not render properly

### ISSUE THREAD

**Issue by xboxdavchi** [microsoft/react-native-windows#4098]
"react-native@0.60.6"
"react-native-windows@0.60.0-vnext.133",
`node -v`: 10.15.0
`npm -v`: 6.4.1

- Target Platform Version(s): 10.0.18362.0
- Target Device(s): Desktop
- Visual Studio Version: VS 2017
- Build Configuration: Debug & Release

### Steps to Reproduce
1. Create a component that has a FlatList in it.
2. Set the inverted property to true

### Expected Behavior
Assuming a list of items 0-100, the FlatList with inverted set to true should render with item 0 at the bottom of the list, and item 100 at the top of the list. 

### Actual Behavior
While the flat list does indeed have the items in the correct order, the whole flat list is flipped on its top edge, meaning that it renders entirely outside of its normal bounds.

**Comment by harinikmsft** [comment]
Assigning to @chrisglein to investigate and figure out the right person to fix this.

**Comment by rectified95** [comment]
Per https://reactnative.dev/docs/flatlist#inverted - the expected behaviour is that the direction of the scroll is reversed.
Also, what do you mean be the list "renders entirely outside of its normal bounds"? The flatlist example in rntester in the Playground app renders correctly - perhaps you could check it out.

**Comment by xboxdavchi** [comment]
That is correct. But it also states how "Uses scale transforms of -1." meaning the list is now also flipped in addition to the direction of the scroll reverse. If you scroll to the top of the react native spec in the example and try out the inverted prop, you see the following:

**EXPECTED**
Normal
![image](https://user-images.githubusercontent.com/37882814/77586261-1b99a480-6ea3-11ea-8298-8856e3d1062d.png)
![image](https://user-images.githubusercontent.com/37882814/77586320-3835dc80-6ea3-11ea-86e2-82a2528636b3.png)

Inverted:
![image](https://user-images.githubusercontent.com/37882814/77586347-4552cb80-6ea3-11ea-968f-bc2116cbec47.png)

**ACTUAL**
Normal: 
![image](https://user-images.githubusercontent.com/37882814/77586474-7f23d200-6ea3-11ea-90f6-51ff815628d3.png)

Inverted: 
![image](https://user-images.githubusercontent.com/37882814/77586556-a67a9f00-6ea3-11ea-80fa-8b0c326dab10.png)

Where the red box is defining the normal bounds of the component. 

I will investigate the playground app and report what I find there...

**Comment by rectified95** [comment]
Thanks for replying.
Here's a screenshot from my PC for reference:
![image](https://user-images.githubusercontent.com/12816515/77587996-33bef300-6ea6-11ea-8e78-dc7d648b2cec.png)

**Comment by xboxdavchi** [comment]
As an additional clarification, I re-verified on the react native spec that the scroll behavior is **not** reversed-- moving the scroll bar up makes the scroll list move up. 

The order in which the items is displayed is reversed though (as expected).

**Comment by rectified95** [comment]
Confirmed, I got the same behaviour as you playing with the React spec editor-demoer. Will investigate. Let me know if you can repro the rendering misplacement issue.

**Comment by rectified95** [comment]
@xboxdavchi Can you confirm that this issue is only about the inverted scroll and that the out-of-bounds rendering problem is longer there?

**Comment by xboxdavchi** [comment]
The out-of-bounds rendering problem still exists. I am not sure how the playground app gets around this, but when I create a flatlist with the inverted property set to true, the behavior is still as I described in my original set of screen shots.

**Comment by xboxdavchi** [comment]
Here is the UI with a flat list. 
![image](https://user-images.githubusercontent.com/37882814/79399650-c51b0580-7f38-11ea-8ee6-cd99ca134a85.png)

Here is the UI with the same flat list, but with the inverted flag set to true: 
![image](https://user-images.githubusercontent.com/37882814/79399570-90a74980-7f38-11ea-9114-f428a2d6c3ae.png)

Notice how the flat list is now rendering out of its normal bounds (it should be at the bottom half of the screen, not flipped over the horizontal axis)

Here is the code snippet for reference: 
`            
<View style={{ flex: 1, justifyContent: "center", alignItems: "center", flexDirection: "column" }}>
                <View style={{ height: 500 }}>
                    <BodyText style={{ color: Colors.White }}>{"flat list test"}</BodyText>
                </View>
                <FlatList
                    style={{ alignSelf: "center", height: 200, width: 200, borderColor: Colors.White, backgroundColor: Colors.Red }}
                    inverted={false}
                    data={myData}
                    renderItem={this.renderItem}>
                </FlatList>
            </View >
`

**Comment by rectified95** [comment]
We believe that the rendering error has been fixed in #4169. Can you upgrade your version of RN Windows or cherry-pick it onto your source version and see if it persists?

As for the scroll direction part, there doesn't seem to be an easy solution to it now. Would you still consider it necessary, if the component rendered properly within its slot?

**Comment by davidpchi** [comment]
Thanks rectified95! 

Unfortunately, the scroll direction part is still necessary for us, as we have a long list of items for this component despite the component rendering properly within its slot. 

We will get back to you asap to confirm that the proposed fix resolves the rendering out of bounds issue.

**Comment by xboxdavchi** [comment]
Thanks rectified95!

Unfortunately, the scroll direction part is still necessary for us, as we have a long list of items for this component despite the component rendering properly within its slot.

We will get back to you asap to confirm that the proposed fix resolves the rendering out of bounds issue.

**Comment by rectified95** [comment]
@xboxdavchi Do you have updates on the rendering part of the problem?

**Comment by xboxdavchi** [comment]
Hey rectified95, 

The issue we are running into now is trying to cherry pick the fix into our branch. We are effectively on 0.60, and the fix in 0.62 diverges enough that the cherry pick is non-trivial. 

We will get back to you as soon as possible once we have figured out how to properly ingest the fix.

**Comment by rectified95** [comment]
@xboxdavchi Do you have updates on your progress?

**Comment by davidpchi** [comment]
Hey Rectified95!

No new progress here, other than we are trying to figure out how to update now into 0.62 from our current fork. 

In the interim, once we finally consume the fix, we'll need the proper scroll behavior regardless, so please keep us posted on that as well! I apologize for the delayed integration!

**Comment by rectified95** [comment]
@davidpchi It looks like the fix on our end will require reimplementing the inversion - as long as the scaleY=-1 transform is passed to XAML under the hood, the scrolling will be incorrect. We're investigating how to model the desired behavior in a new way.

Can you tell me what your usage scenario is in more detail? 
For example:
- whether it is about having a chat-like experience, where you will be adding new entries, and want them to appear at the bottom (would you want this addition to cause a slight scroll downwards as well, to track the latest 'message'?)
- will you be toggling the inverted property after your list has been rendered? If so, would you then want to see the first element rendered at the bottom, or would you be seeing the last element at the top, without the list still showing the first element, now at the bottom.

**Comment by NickGerleman** [comment]
This doesn't seem within the scope of the initial 0.62 release. Flagging for retriage. We should backport this to 0.62 after release as its non-breaking and there's a customer need.

**Comment by xboxdavchi** [comment]
Hey Rectified95, 

The usage scenario in this case is a chat-like experience (the first one you described). We will not toggle the inverted property after our list has been renedered.

**Comment by chrisglein** [comment]
> This doesn't seem within the scope of the initial 0.62 release. Flagging for retriage. We should backport this to 0.62 after release as its non-breaking and there's a customer need.

It's a high profile partner bug. After many weeks of digging in, it looks like we need to re-architect how we implement inversion on FlatList, so I have low confidence this is trending towards the 0.62 release. But it is important.

Going to keep aspirationally (we would backport), but downgrading "must have".

**Comment by chrisglein** [comment]
If we need to flip the mouse wheel direction and the platform doesn't support this, then this may turn into a WinUI3 ask. @rectified95 let us know what recommendation you end up making here.

**Comment by rectified95** [comment]
I'm investigating an alternative solution that would involve writing a custom panel for the XAML ScrollViewer we're using. Will report when I know more. 

If we could just intercept the mouse delta, that'd make things easy on JS side, though.

**Comment by rectified95** [comment]
This needs a change in XAML to work properly.

**Comment by siddhant-rigi** [comment]
Can't RN release an update to fix this issue forever.

**Comment by ClaudiuHBann** [comment]
Any new updates on this?
